File: app/src/main/java/com/example/aman/ambedkarmission/Adapter/StateAdapter.java
package com.example.aman.ambedkarmission.Adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ListView;
import android.widget.TextView;

import com.example.aman.ambedkarmission.R;

import java.util.ArrayList;

public class StateAdapter extends BaseAdapter {
    Context context;
    ArrayList<String> arrayLists;
    LayoutInflater layoutInflater;
    TextView wwe;
    ListView listViews;
    public StateAdapter(Context applicationContext, ArrayList<String> arrayList, TextView state, ListView listView) {
        context=applicationContext;
        arrayLists=arrayList;
        layoutInflater=LayoutInflater.from(context);
        wwe=state;
        listViews=listView;

    }

    @Override
    public int getCount() {
        return arrayLists.size();
    }

    @Override
    public Object getItem(int i) {
        return arrayLists.get(i);
    }

    @Override
    public long getItemId(int i) {
        return i;
    }

    @Override
    public View getView(final int i, View view, ViewGroup viewGroup) {
        view=layoutInflater.inflate(R.layout.list_layout,null);
        TextView text;
        text=(TextView)view.findViewById(R.id.text);
        text.setText(arrayLists.get(i));
        text.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                wwe.setText(arrayLists.get(i));
                listViews.setVisibility(View.GONE);

            }
        });
        return view ;
    }
}

File: app/src/main/java/com/info/work/ambedkarmission/Adapter/StateAdapter.kt
package com.ois.todo.countrylist

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.TextView
import com.info.work.ambedkarmission.R
import com.info.work.ambedkarmission.model.State

import com.ois.todo.delegate.CountrySelectListener


class StateAdapter(internal var mContext: Context?, private val stateList: List<State>) : RecyclerView.Adapter<StateAdapter.MyViewHolder>() {
    internal lateinit var countrySelectListener: CountrySelectListener

    fun onclickListener(countrySelectListener: CountrySelectListener) {
        this.countrySelectListener = countrySelectListener
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val itemView = LayoutInflater.from(parent.context)
                .inflate(R.layout.dialog_select_country_code_raw, parent, false)

        return MyViewHolder(itemView)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        val country = stateList[position]
        if (country != null) {
            holder.txtCountryName.text = country.state
           // holder.txtCountryCode.text = country.countryCodeStr
           // holder.imgCountryFlag.setImageResource(country.resId)
        }

    }

    override fun getItemCount(): Int {
        return stateList.size
    }

    inner class MyViewHolder(view: View) : RecyclerView.ViewHolder(view), View.OnClickListener {
        //var imgCountryFlag: ImageView
        var txtCountryName: TextView
       // var txtCountryCode: TextView
        var llCountryRow: LinearLayout

        init {

            //imgCountryFlag = view.findViewById<View>(R.id.imgCountryFlag) as ImageView
            txtCountryName = view.findViewById<View>(R.id.txtCountryName) as TextView
            //txtCountryCode = view.findViewById<View>(R.id.txtCountryCode) as TextView
            llCountryRow = view.findViewById<View>(R.id.llCountryRow) as LinearLayout
            llCountryRow.setOnClickListener(this)

        }

        override fun onClick(v: View) {
            when (v.id) {
                R.id.llCountryRow -> countrySelectListener.onCountryClick(adapterPosition)
            }
        }
    }
}